### ISSUE TITLE
dual SIM phones

### ISSUE THREAD

**Issue by ankek** [SilenceIM/Silence#661]
### How to reproduce
- I can send and receive SMS only from 1 SIM card. The second SIM is unavailable. I can't read any messages for that. And I can't send messages from second SIM. 

**Actual result:** 
**Expected result:** 

### Device info
- Device: Moto g5s
- Android version: 7.1
- Silence version: last

**Comment by silkshadow** [comment]
Just to add detail or possible another aspect to this.

In Silence, if I send a text message to a recipient, who is also using Silence, from SIM 1 and do the key exchange, there is no problem. If I then send a message from SIM 2 to the same recipient, they cannot see the message and it reports as "Message encrypted for non-existing session". 

If I do a key exchange from SIM 2 then it works, but any messages sent from SIM1 the recipient cannot see and it reports as "Message encrypted for non-existing session". 

In other words, on the receiving end, it seems maybe Silence cannot separate sessions from different originating SIMs.

Silence 0.15.13 and LineageOS 14.1 (on both ends). 

Thanks!

**Comment by tuxayo** [comment]
related #467

**Comment by nailyk-fr** [comment]
Same issue with answering. The card to send should be the last received, or an option into the conversation settings. 

Sorry to bother you here, but gone to IRC and didn't get answers, so I copy/paste here: 

> nailyk | hello. Just had a talk with xxxxxx which said me this is difficult for you to debug dual sims problem on silence because you do not have dual sim device.  I would be happy to help you on this. 
>  Already have my fdroid repo, so building & testing Silence (from master or another branch) shouldn't be a problem
